### PR TITLE
add follow option to generate + sort accounts on list

### PR DESCRIPTION
**Pull request by obouchet**
### Description

Presently, a call to `generate` command returns the current `totp` code and exits. 
This PR is a proposal to add a `follow` parameter for the `generate` command so that the `generate` command continues generating `totp` code until stopped by the user.
The code is then generated every second and only printed on the console whenever it changes.

This PR also adds the sorting of accounts (by names) on the `list` command for convenience.

### CODE REVIEW

File: internal/cmd/generate.go [hunk 1]
```diff
@@ -30,6 +30,12 @@ func (c *Generate) Execute(opts *commander.CommandHelper) {
 		return
 	}
 
+	mustFollow := false
+	follow := opts.Arg(2)
```
Comment (yitsushi) [inline internal/cmd/generate.go:34]: Please add it as a flag and not a positional argument.

Example:
* Configure: https://github.com/yitsushi/totp-cli/blob/main/internal/cmd/set_prefix.go#L91-L93
* In action: https://github.com/yitsushi/totp-cli/blob/main/internal/cmd/set_prefix.go#L125-L127

File: internal/cmd/generate.go [hunk 2]
```diff
@@ -48,6 +54,23 @@ func (c *Generate) Execute(opts *commander.CommandHelper) {
...
+		previousCode := ""
+		for {
+			code := generateCode(account)
+			if code != previousCode {
+				fmt.Println(code)
+				previousCode = code
+			}
+			time.Sleep(time.Second)
+		}
+	}
```
Comment (yitsushi) [inline internal/cmd/generate.go:70]:
```suggestion
	if !mustFollow {
		code := generateCode(account)
		fmt.Println(code)

		return
	}
	
	previousCode := ""
	for {
		code := generateCode(account)
		if code != previousCode {
			fmt.Println(code)
			previousCode = code
		}
		time.Sleep(time.Second)
	}
```

Comment (yitsushi) [reply internal/cmd/generate.go:70]: Or even remove the whole extra generate call:

```suggestion
	code := generateCode(account)
	fmt.Println(code)
	
	if !mustFollow {
		return
	}
	
	previousCode := code
	for {
		code := generateCode(account)
		if code != previousCode {
			fmt.Println(code)
			previousCode = code
		}
		time.Sleep(time.Second)
	}
```

File: internal/cmd/list.go [hunk 3]
```diff
@@ -35,6 +36,10 @@ func (c *List) Execute(opts *commander.CommandHelper) {
 		os.Exit(1)
 	}
 
+	sort.Slice(namespace.Accounts, func(i, j int) bool {
```
Comment (yitsushi) [inline internal/cmd/list.go:39]: Sorting before save (on add/update/delete) would be more efficient, but does not add much cost here unless someone has 2million accounts in one namespace (they shouldn't), so fine by me.

**Review by yitsushi — CHANGES_REQUESTED** [review]
Overall looks good, a few changes:
1. Move convert the `follow` argument into a flag.
2. Add documentation into the Readme file please.

The code suggestion, I like when I can get rid of extra branches. If I can't drop an `else` branch, usually I stop and re-think what I'm doing.

File: internal/cmd/generate.go (hunk 1 above)
Comment (obouchet) [reply internal/cmd/generate.go]: Thank you for pointing me out to the flag possibility. It is much better.

File: internal/cmd/generate.go (hunk 2 above)
Comment (obouchet) [reply internal/cmd/generate.go]: I've gone with the last suggestion. much cleaner indeed. Thanks!

**Review by yitsushi — APPROVED** [review]

**Comment by yitsushi** [comment]
Looking into the linter check locally, somehow the linter output in the workflow is useless.

**Comment by obouchet** [comment]
> Looking into the linter check locally, somehow the linter output in the workflow is useless.

I'll check that right now.

File: internal/cmd/generate.go [hunk 4]
```diff
@@ -48,6 +50,25 @@ func (c *Generate) Execute(opts *commander.CommandHelper) {
...
 
+	code := generateCode(account)
+	fmt.Println(code)
+
+	if !mustFollow {
+		return
+	}
+
+	previousCode := code
+	for {
```
Comment (yitsushi) [inline internal/cmd/generate.go:61]:
```suggestion

	for {
```

File: internal/cmd/generate.go [hunk 5]
```diff
@@ -48,6 +50,25 @@ func (c *Generate) Execute(opts *commander.CommandHelper) {
...
+		code := generateCode(account)
+		if code != previousCode {
+			fmt.Println(code)
+			previousCode = code
+		}
+		time.Sleep(time.Second)
```
Comment (yitsushi) [inline internal/cmd/generate.go:67]:
```suggestion

		time.Sleep(time.Second)
```

File: internal/cmd/generate.go [hunk 6]
```diff
@@ -56,14 +77,16 @@ func (c *Generate) Execute(opts *commander.CommandHelper) {
 	if account.Prefix != "" {
 		code = account.Prefix + code
 	}
-
-	fmt.Println(code)
+	return code
```
Comment (yitsushi) [inline internal/cmd/generate.go:80]:
```suggestion

	return code
```

**Review by yitsushi — COMMENTED** [review]
These are the issues

**Comment by obouchet** [comment]
@yitsushi Thank you very much!

**Pull request merged**